lib/CPAN2OBS.pm

```
@@ -0,0 +1,853 @@
+    }
+    if (not -s $letter_xml) {
+        return \%obs_info;
+    }
+    my $info = XMLin($letter_xml)->{sourceinfo};
+    return \%obs_info unless $info;
+    $info = [$info] unless ref $info eq 'ARRAY';
+
+    mkdir "$data/project-xml";
+    for my $pi (@$info) {
+        my $srcmd5 = $pi->{srcmd5};
+        my $package = $pi->{package};
+        my $cached = $cache->{ $srcmd5 };
+        if (not defined $cached) {
+            info("fetch obs xml for $package");
+            my $cmd = sprintf "osc -A %s api /source/%s/%s >$data/project-xml/$package.xml",
+                $apiurl, $project_prefix, $package;
+            debug("CMD $cmd");
+            system $cmd;
+
+            unless (-s "$data/project-xml/$package.xml") {
+                warn __PACKAGE__.':'.__LINE__.": !!!! no osc data for $package\n";
+                next;
+            }
+            my $pxml = XMLin "$data/project-xml/$package.xml";
+            my $name = $pxml->{name};
+            $obs_info{ $name }->{ok} = 1;
+            if ($pxml->{entry}->{'cpanspec.error'}) {
+                $obs_info{ $name }->{ok} = 0;
+            }
+            for my $entry (keys %{ $pxml->{entry} }) {
+                if ($entry =~ m/\.tar/ || $entry =~ m/\.tgz$/ || $entry =~ m/\.zip$/) {
+                    $obs_info{ $name }->{archive} = $entry;
+                }
+            }
+            $cache->{ $srcmd5 } = $obs_info{ $package } || {};
+            $self->store_obs_cache($letter, $cache);
+        }
+        else {
+            if (ref $cached) {
+                $obs_info{ $package } = $cached;
+            }
+            else {
+                $obs_info{ $package } = { archive => $cached, ok => 1 };
+            }
+        }
+    }
+    return \%obs_info;
+}
+
+sub fetch_obs_cache {
+    my ($self, $letter) = @_;
+    my $data = $self->data;
+    mkdir "$data/obs-cache";
+    my $cachefile = "$data/obs-cache/$letter";
+    my $cache = {};
+    eval { $cache = retrieve($cachefile); };
```

Do you really mean to ignore all exceptions here?